File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/dtos/ProblemMetadata.java
package io.github.sanyavertolet.edukate.backend.dtos;

import io.github.sanyavertolet.edukate.backend.entities.Problem;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.With;
import org.springframework.data.annotation.PersistenceCreator;

import java.util.List;

@Data
@AllArgsConstructor(onConstructor = @__(@PersistenceCreator))
public class ProblemMetadata {
    private String name;
    private Boolean isHard;
    private List<String> tags;
    @With
    private Problem.Status status;
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/dtos/ProblemMetadata.kt
package io.github.sanyavertolet.edukate.backend.dtos

import io.github.sanyavertolet.edukate.backend.entities.Problem
import io.github.sanyavertolet.edukate.common.ContentLanguage
import java.time.Instant

data class ProblemMetadata(
    val key: String,
    val code: String,
    val bookSlug: String,
    val isHard: Boolean,
    val tags: List<String>,
    val status: Problem.Status,
    val createdAt: Instant,
    val language: ContentLanguage,
)
